src/Security/Group.php

```diff
@@ -327,9 +327,16 @@ public function Members($filter = '')
                 $query->removeFilterOn('Group_Members');
             });
         }
+
         // Now set all children groups as a new foreign key
-        $groups = Group::get()->byIDs($this->collateFamilyIDs());
-        $result = $result->forForeignID($groups->column('ID'))->where($filter);
+        $familyIDs = $this->collateFamilyIDs();
+        if (!empty($familyIDs)) {
+            $groups = Group::get()->byIDs($familyIDs);
+            $groupIDs = $groups->column('ID');
+            if (!empty($groupIDs)) {
```

Comment:
This question is related to existing code, but why do we query groups based on a list of ids, then just get the ID column from that result? Isn't that just another unnecessary SQL query that's as simple as `$groupIDs = $familyIDs`?

Reply:
I'm going to find out why, by messing around with it. :)